ms2/src/org/labkey/ms2/metadata/MassSpecFractionsDomainKind.java

```diff
@@ -122,8 +125,19 @@ public void appendNavTrail(NavTree root, Container c, User user)
     }
 
     @Override
-    public Domain createDomain(GWTDomain domain, JSONObject arguments, Container container, User user, @Nullable TemplateInfo templateInfo)
+    public Domain createDomain(GWTDomain domain, SampleTypeDomainKindProperties arguments, Container container, User user, @Nullable TemplateInfo templateInfo)
     {
-        return _assayDelegate.createDomain(domain, arguments, container, user, templateInfo);
+        //TODO this is probably broken.
```

Comment:
@labkey-ians Is this TODOs still valid or have you verified these changes? 

@labkey-jeckels Josh, can you review this PR as it relates to the PR changes that Binal made for the other domain kinds?